lms/djangoapps/grades/rest_api/v1/tests/test_gradebook_views.py

```diff
@@ -1280,6 +1288,102 @@ def test_filter_course_grade_absent_without_min(self, login_method, num_enrollme
                 assert actual_data['total_users_count'] == num_enrollments
                 assert actual_data['filtered_users_count'] == num_enrollments
 
+    @contextmanager
+    def _mock_all_course_grade_reads(self, percent=0.9):
+        """
+        A context manager for mocking CourseGradeFactory.read and returning the same grade for all learners.
+        """
+        # pylint: disable=unused-argument
+        def fake_course_grade_read(*args, **kwargs):
+            return self.mock_course_grade(kwargs['user'], passed=True, percent=percent)
+
+        with patch('lms.djangoapps.grades.course_grade_factory.CourseGradeFactory.read') as mock_read:
+            mock_read.side_effect = fake_course_grade_read
+            yield
+
+    def _assert_usernames(self, response, expected_usernames):
+        """ Helper method to assert that the expected users were returned from the endpoint """
+        assert status.HTTP_200_OK == response.status_code
+        response_data = dict(response.data)
+        actual_usernames = [row['username'] for row in response_data['results']]
+        assert set(actual_usernames) == set(expected_usernames)
+
+    def test_users_with_course_roles(self):
+        """ Test that a staff member erolled in the course will be included in grade results. """
+        # This function creates and enrolls a course staff (not global staff) user
+        staff_user = self.login_course_staff()
+        with override_waffle_flag(self.waffle_flag, active=True):
+            with self._mock_all_course_grade_reads():
+                response = self.client.get(self.get_url(course_key=self.course.id))
+        self._assert_usernames(response, [
```

Comment:
Proposed test cleanup: collecting those students into an array for readability/reuse:

```
self.course_students = [ self.student.username, self.other_student.username, ...]
self._assert_usernames(response, self.course_students + staff_user.username)
```